In [1]:
%matplotlib inline
%load_ext autoreload
%autoreload 2

import sys,os
import numpy as np
import matplotlib.pyplot as plt

sys.path.append("../run/")
from run import run_scan

## Float $N_D$, $z_0$ and $\beta$

In [2]:
fixed_params = ['n','sigma','alpha','Lmax_disk', 'Lmax_bulge','N_bulge']
fixed_param_vals = [2.35,1.528,2.6,1.0e36,1.0e36,0]

floated_params = ['N_disk', 'z0', 'beta']
floated_param_priors = [[0,2000000],[0.01,2.0],[1.1,2.5]]

In [3]:
rs_nd = run_scan(fixed_params, fixed_param_vals, floated_params, floated_param_priors, Ns = 200, Nang = 1, share_betas=True, use_prior=False)

In [4]:
rs_nd.perform_scan_minuit()

../run/run.py:180: RuntimeWarning: fcn returns Nan
fcn is called with following arguments:
    N_disk = +1036812.618863
        z0 = +1.860780
      beta = +2.325676

  self.m.migrad()
../run/run.py:180: RuntimeWarning: fcn returns Nan
fcn is called with following arguments:
    N_disk = +nan
        z0 = +nan
      beta = +nan

  self.m.migrad()
../run/run.py:180: RuntimeWarning: fcn returns Nan
fcn is called with following arguments:
    N_disk = +1205194.903950
        z0 = +0.080480
      beta = +2.006594

  self.m.migrad()
../run/run.py:180: RuntimeWarning: fcn returns Nan
fcn is called with following arguments:
    N_disk = +1211580.232884
        z0 = +0.078898
      beta = +2.005235

  self.m.migrad()
../run/run.py:180: RuntimeWarning: fcn returns Nan
fcn is called with following arguments:
    N_disk = -nan
        z0 = -nan
      beta = -nan

  self.m.migrad()
../run/run.py:180: RuntimeWarning: fcn returns Nan
fcn is called with following arguments:
    N_disk = +1217371.9322

FCN = 293.403706173 
 TOTAL NCALL = 408 
 NCALLS = 408 
 
 
 EDM = 0.0316960164702 
 GOAL EDM = 1e-05 
 
 UP = 1.0 
 
 
 
 
 
 Valid 
 Valid Param 
 Accurate Covar 
 PosDef 
 Made PosDef 
 
 
 False 
 True 
 True 
 True 
 False 
 
 
 Hesse Fail 
 HasCov 
 Above EDM 
 
 Reach calllim 
 
 
 False 
 True 
 True 
 
 False

+ 
 Name 
 Value 
 Parab Error 
 Minos Error- 
 Minos Error+ 
 Limit- 
 Limit+ 
 FIXED 
 
 
 
 1 
 N_disk 
 1.12779e+06 
 494265 
 0 
 0 
 0.0 
 2000000.0 
 
 
 
 
 2 
 z0 
 0.0856853 
 0.0566937 
 0 
 0 
 0.01 
 2.0 
 
 
 
 
 3 
 beta 
 1.99518 
 0.0799758 
 0 
 0 
 1.1 
 2.5 
 
 
 
 
 
 
 \begin{tabular}{|c|r|r|r|r|r|r|r|c|}
\hline
 & Name & Value & Para Error & Error+ & Error- & Limit+ & Limit- & FIXED\\
\hline
1 & $N_{disk}$ & 1.128e+06 & 4.943e+05 & & & 0.000e+00 & 2.000e+06 & \\
\hline
2 & z0 & 8.569e-02 & 5.669e-02 & & & 1.000e-02 & 2.000e+00 & \\
\hline
3 & $\beta$ & 1.995e+00 & 7.998e-02 & & & 1.100e+00 & 2.500e+00 & \\
\hline
\end{tabular}

## Float $N_D$, $N_B$, $z_0$ and $\beta$

In [5]:
fixed_params = ['n','sigma','alpha','Lmax_disk', 'Lmax_bulge']
fixed_param_vals = [2.35,1.528,2.6,1.0e36,1.0e36]

floated_params = ['N_bulge','N_disk','z0', 'beta']
floated_param_priors = [[0,2000000],[0,2000000],[0.01,2.0],[1.1,2.0]]

In [6]:
rs_nd_nb = run_scan(fixed_params, fixed_param_vals, floated_params, floated_param_priors, Ns = 400, Nang = 5, share_betas=True, use_prior=False)

In [7]:
rs_nd_nb.perform_scan_minuit()

FCN = 291.428801732 
 TOTAL NCALL = 481 
 NCALLS = 481 
 
 
 EDM = 4.50073218407e-06 
 GOAL EDM = 1e-05 
 
 UP = 1.0 
 
 
 
 
 
 Valid 
 Valid Param 
 Accurate Covar 
 PosDef 
 Made PosDef 
 
 
 True 
 True 
 True 
 True 
 False 
 
 
 Hesse Fail 
 HasCov 
 Above EDM 
 
 Reach calllim 
 
 
 False 
 True 
 False 
 
 False

+ 
 Name 
 Value 
 Parab Error 
 Minos Error- 
 Minos Error+ 
 Limit- 
 Limit+ 
 FIXED 
 
 
 
 1 
 N_bulge 
 265164 
 117051 
 0 
 0 
 0.0 
 2000000.0 
 
 
 
 
 2 
 N_disk 
 675207 
 174537 
 0 
 0 
 0.0 
 2000000.0 
 
 
 
 
 3 
 z0 
 0.0665051 
 0.0421331 
 0 
 0 
 0.01 
 2.0 
 
 
 
 
 4 
 beta 
 2 
 0.142472 
 0 
 0 
 1.1 
 2.0 
 
 
 
 
 
 
 \begin{tabular}{|c|r|r|r|r|r|r|r|c|}
\hline
 & Name & Value & Para Error & Error+ & Error- & Limit+ & Limit- & FIXED\\
\hline
1 & $N_{bulge}$ & 2.652e+05 & 1.171e+05 & & & 0.000e+00 & 2.000e+06 & \\
\hline
2 & $N_{disk}$ & 6.752e+05 & 1.745e+05 & & & 0.000e+00 & 2.000e+06 & \\
\hline
3 & z0 & 6.651e-02 & 4.213e-02 & & & 1.000e-02 & 2.000e+00 & \\
\hline
4 & $\beta$ & 2.000e+00 & 1.425e-01 & & & 1.100e+00 & 2.000e+00 & \\
\hline
\end{tabular}

In [8]:
TS = 2*(rs_nd_nb.max_LL - rs_nd.max_LL)
print "TS = ", TS
print np.sqrt(TS), "sigma"

TS =  11.4362383201
3.38175077735 sigma


## Float $N_D$, $N_B$, $z_0$, $\beta$ and $\alpha$

In [56]:
fixed_params = ['n','sigma','Lmax_disk', 'Lmax_bulge']
fixed_param_vals = [2.35,1.528,1.0e36,1.0e36]

floated_params = ['N_bulge','N_disk', 'alpha','z0', 'beta']
floated_param_priors = [[0,3000000],[0,3000000],[2.1,3.0],[0.01,2.0],[1.1,2.0]]

In [57]:
rs_nd_nb_alpha = run_scan(fixed_params, fixed_param_vals, floated_params, floated_param_priors, Ns = 200, Nang = 5, share_betas=True, use_prior=True)

In [58]:
rs_nd_nb_alpha.perform_scan_minuit()

FCN = 289.427712929 
 TOTAL NCALL = 786 
 NCALLS = 786 
 
 
 EDM = 0.000121131845225 
 GOAL EDM = 1e-05 
 
 UP = 1.0 
 
 
 
 
 
 Valid 
 Valid Param 
 Accurate Covar 
 PosDef 
 Made PosDef 
 
 
 True 
 True 
 True 
 True 
 False 
 
 
 Hesse Fail 
 HasCov 
 Above EDM 
 
 Reach calllim 
 
 
 False 
 True 
 False 
 
 False

+ 
 Name 
 Value 
 Parab Error 
 Minos Error- 
 Minos Error+ 
 Limit- 
 Limit+ 
 FIXED 
 
 
 
 1 
 N_bulge 
 849551 
 2.53521e+06 
 0 
 0 
 0.0 
 3000000.0 
 
 
 
 
 2 
 N_disk 
 869369 
 426897 
 0 
 0 
 0.0 
 3000000.0 
 
 
 
 
 3 
 alpha 
 2.93639 
 0.658019 
 0 
 0 
 2.1 
 3.0 
 
 
 
 
 4 
 z0 
 0.0787013 
 0.0577808 
 0 
 0 
 0.01 
 2.0 
 
 
 
 
 5 
 beta 
 1.99745 
 0.725188 
 0 
 0 
 1.1 
 2.0 
 
 
 
 
 
 
 \begin{tabular}{|c|r|r|r|r|r|r|r|c|}
\hline
 & Name & Value & Para Error & Error+ & Error- & Limit+ & Limit- & FIXED\\
\hline
1 & $N_{bulge}$ & 8.496e+05 & 2.535e+06 & & & 0.000e+00 & 3.000e+06 & \\
\hline
2 & $N_{disk}$ & 8.694e+05 & 4.269e+05 & & & 0.000e+00 & 3.000e+06 & \\
\hline
3 & $\alpha$ & 2.936e+00 & 6.580e-01 & & & 2.100e+00 & 3.000e+00 & \\
\hline
4 & z0 & 7.870e-02 & 5.778e-02 & & & 1.000e-02 & 2.000e+00 & \\
\hline
5 & $\beta$ & 1.997e+00 & 7.252e-01 & & & 1.100e+00 & 2.000e+00 & \\
\hline
\end{tabular}

In [60]:
TS = 2*(rs_nd_nb_alpha.max_LL - rs_nd.max_LL)
print "TS = ", TS
print np.sqrt(TS), "sigma"

TS =  15.4384159268
3.9291749677 sigma


## Float $N_D$ 

In [12]:
fixed_params = ['n','sigma','alpha','beta_bulge','beta_disk','z0','N_bulge','Lmax_disk', 'Lmax_bulge']
fixed_param_vals = [2.35,1.528,2.6,1.2,1.2, 0.7,0,1.0e36,1.0e36]

floated_params = ['N_disk']
floated_param_priors = [[0,10000]]

In [6]:
rs_nd_only = run_scan(fixed_params, fixed_param_vals, floated_params, floated_param_priors, Ns = 200, Nang = 5, use_prior=True)

In [7]:
rs_nd_only.perform_scan_minuit()

FCN = 331.153637874 
 TOTAL NCALL = 22 
 NCALLS = 22 
 
 
 EDM = 3.07013101117e-10 
 GOAL EDM = 1e-05 
 
 UP = 1.0 
 
 
 
 
 
 Valid 
 Valid Param 
 Accurate Covar 
 PosDef 
 Made PosDef 
 
 
 True 
 True 
 True 
 True 
 False 
 
 
 Hesse Fail 
 HasCov 
 Above EDM 
 
 Reach calllim 
 
 
 False 
 True 
 False 
 
 False

+ 
 Name 
 Value 
 Parab Error 
 Minos Error- 
 Minos Error+ 
 Limit- 
 Limit+ 
 FIXED 
 
 
 
 1 
 N_disk 
 4462.95 
 635.879 
 0 
 0 
 0.0 
 10000.0 
 
 
 
 
 
 
 \begin{tabular}{|c|r|r|r|r|r|r|r|c|}
\hline
 & Name & Value & Para Error & Error+ & Error- & Limit+ & Limit- & FIXED\\
\hline
1 & $N_{disk}$ & 4.463e+03 & 6.359e+02 & & & 0.000e+00 & 1.000e+04 & \\
\hline
\end{tabular}

## Float $N_D$ and $N_B$

In [8]:
fixed_params = ['n','sigma','alpha','beta_bulge','beta_disk','z0','Lmax_disk', 'Lmax_bulge']
fixed_param_vals = [2.35,1.528,2.6,1.2,1.2, 0.7,1.0e36,1.0e36]

floated_params = ['N_bulge','N_disk']
floated_param_priors = [[0,8000],[0,10000]]

In [9]:
rs_nd_nb_only = run_scan(fixed_params, fixed_param_vals, floated_params, floated_param_priors, Ns = 200, Nang = 5, use_prior=True)

In [10]:
rs_nd_nb_only.perform_scan_minuit()

FCN = 326.689605639 
 TOTAL NCALL = 77 
 NCALLS = 77 
 
 
 EDM = 3.41620193571e-05 
 GOAL EDM = 1e-05 
 
 UP = 1.0 
 
 
 
 
 
 Valid 
 Valid Param 
 Accurate Covar 
 PosDef 
 Made PosDef 
 
 
 True 
 True 
 True 
 True 
 False 
 
 
 Hesse Fail 
 HasCov 
 Above EDM 
 
 Reach calllim 
 
 
 False 
 True 
 False 
 
 False

+ 
 Name 
 Value 
 Parab Error 
 Minos Error- 
 Minos Error+ 
 Limit- 
 Limit+ 
 FIXED 
 
 
 
 1 
 N_bulge 
 690.108 
 359.019 
 0 
 0 
 0.0 
 8000.0 
 
 
 
 
 2 
 N_disk 
 3152.13 
 800.333 
 0 
 0 
 0.0 
 10000.0 
 
 
 
 
 
 
 \begin{tabular}{|c|r|r|r|r|r|r|r|c|}
\hline
 & Name & Value & Para Error & Error+ & Error- & Limit+ & Limit- & FIXED\\
\hline
1 & $N_{bulge}$ & 6.901e+02 & 3.590e+02 & & & 0.000e+00 & 8.000e+03 & \\
\hline
2 & $N_{disk}$ & 3.152e+03 & 8.003e+02 & & & 0.000e+00 & 1.000e+04 & \\
\hline
\end{tabular}

In [13]:
TS = 2*(rs_nd_nb_only.max_LL - rs_nd_only.max_LL)
print "TS = ", TS
print np.sqrt(TS), "sigma"

TS =  8.92806446994
2.98798669173 sigma


# Table

Unless floated, $n = 2.35$,
$\sigma = 1.528$,
$\alpha = 2.6$,
$\beta = 1.2$,
$z_0 = 0.7$.

Only floating $N_D$ and $N_B$:

| $N_D$        | $N_B$       | TS   |
|--------------|-------------|------|
| $4462\pm635$ | 0           | 0    |
| $3152\pm800$ | $690\pm359$ | 9    |


Floating other paramsters as in 2FIG Table 2:


| $N_D$             | $N_B$                     | $z_0$         | $\beta$       | $\alpha$      | TS |
|-------------------|---------------------------|---------------|---------------|---------------|----|
| $953805\pm149704$ | 0                         | $0.11\pm0.05$ | $2\pm0.08$    | -             | 0  |
| $675207\pm174537$ | $265164\pm117051$         | $0.07\pm0.04$ | $2\pm0.14$    | -             | 11 |
| $869369\pm426897$ | $849551\pm2.53\times10^6$ | $0.08\pm0.06$ | $1.99\pm0.72$ | $2.93\pm0.66$ | 15 |
